src/client.c

```diff
@@ -35,6 +35,51 @@ static const char *hdr2hex(const struct MBuf *data, char *buf, unsigned buflen)
 	return bin2hex(bin, dlen, buf, buflen);
 }
 
+/*
+ * An helper function to select authentication database for the current
+ * client. The order of preference will be as follows:
+ *   1) client->db->auth_dbname: Client's target database may alraedy have
+ * 		an auth_database defined. Prefer this when available.
+ *   2) cf_auth_dbname: Global authentication database name.
+ *   3) client->db: Client's target database. 
+ * NOTE: If authentication database is found but it is disabled, then
+ * 		 client will get disconnected. 
+ * Returns:
+ *   auth_database, based on the above preference.
+ */
+PgDatabase *prepare_auth_database(PgSocket *client)
```

Comment:
I think the many nested ifs make this function quite hard to read. I think readability here would benefit a lot from early returns. pseudo code:
```
if (!client->db->auth_dbname && !cf_auth_dbname)
   return client->db

auth_dbname = client->db->auth_dbname ? client->db->auth_dbname : cf_auth_dbname


auth_db = find_database(auth_dbname)
if (!auth_db)
    return client->db

if auth_db->db_disabled
    disconnect_client
    return NULL
return auth_db;
```

Reply:
Thank you for the suggestion. Updated the code.